test/specs/putTotalPages.spec.js

```diff
@@ -42,3 +42,46 @@ describe("Module: putTotalPages", () => {
     comparePdf(doc.output(), "customfont.pdf", "putTotalPages");
   });
 });
+
+it("customfont with encoding without passing fontWeight", () => {
+  var PTSans = loadBinaryResource("reference/PTSans.ttf");
+  var doc = new jsPDF({ filters: ["ASCIIHexEncode"], floatPrecision: 2 });
+  var totalPagesExp = "{totalPages}";
+
+  doc.addFileToVFS("PTSans.ttf", PTSans);
+  doc.addFont("PTSans.ttf", "PTSans", "normal" , "Identity-H");
+
+  doc.setFont("PTSans");
+
+  doc.text(10, 10, "Page 1 of {totalPages}");
+  doc.addPage();
+
+  doc.text(10, 10, "Page 2 of {totalPages}");
+
+  if (typeof doc.putTotalPages === "function") {
+    doc.putTotalPages(totalPagesExp);
+  }
+
+  comparePdf(doc.output(), "customfont.pdf", "putTotalPages");
+});
+
+it("customfont with fontweight", () => {
+  var PTSans = loadBinaryResource("reference/PTSans.ttf");
+  var doc = new jsPDF({ filters: ["ASCIIHexEncode"], floatPrecision: 2 });
+  var totalPagesExp = "{totalPages}";
+
+  doc.addFileToVFS("PTSans.ttf", PTSans);
+  doc.addFont("PTSans.ttf", "PTSans", "normal",200, "Identity-H");
+
+  doc.setFont("PTSans",'normal200');
```

Comment:
I think we should change `setFont`, as well.

Reply:
by adding a fontweight param with that too? @HackbrettXXX